test/nni_search.py

```diff
@@ -0,0 +1,1454 @@
+        return tree_inst, trees
+
+    @staticmethod
+    def load_dag(fasta_path, newick_path):
+        rand = random.randint(10000, 99999)
+        dag_inst = bito.gp_instance(f"_ignore/mmap.${rand}.data")
+        dag_inst.read_fasta_file(fasta_path)
+        # dag_inst.read_newick_file(newick_path)
+        dag_inst.read_newick_file(newick_path, sort_taxa)
+        dag_inst.make_dag()
+        dag = dag_inst.get_dag()
+        return dag_inst, dag
+
+    @staticmethod
+    def load_pps(pp_path):
+        pps = []
+        with open(pp_path, 'r') as fp:
+            for line in fp.readlines():
+                pps.append(float(line))
+        return pps
+
+    @staticmethod
+    def load_pcsp_pp_map(pcsp_pp_path):
+        pcsp_pp_map = {}
+        df = pd.read_csv(pcsp_pp_path)
+        for index, row in df.iterrows():
+            parent = row["parent"].split("|")
+            parent = bito.subsplit(parent[0], parent[1])
+            child = row["child"].split("|")
+            child = bito.subsplit(child[0], child[1])
+            pcsp = bito.pcsp(parent, child)
+            pp = float(row["pcsp_pp"])
+            pcsp_pp_map[pcsp] = pp
+        return pcsp_pp_map
+
+    @staticmethod
+    def build_tree_id_map(trees):
+        tree_id_map = {}
+        for id, tree in enumerate(trees):
+            tree_id_map[id] = tree
+        return tree_id_map
+
+    @staticmethod
+    def build_tree_pp_map(tree_id_map, pps):
+        tree_pp_map = {}
+        for (tree_id, pp) in zip(tree_id_map, pps):
+            tree_pp_map[tree_id] = pp
+        return tree_pp_map
+
+    @staticmethod
+    def build_pcsp_pp_map(dag, tree_id_map, tree_pp_map):
+        dag_pcsps = dag.build_set_of_edge_bitsets()
+        pcsp_pp_map = {}
+        for pcsp_count, pcsp in enumerate(dag_pcsps):
+            print_v(f"# loading pcsp {pcsp_count} of {len(dag_pcsps)}...")
+            pcsp_pp_map[pcsp] = 0.0
+        for tree_id in tree_pp_map:
+            tree = tree_id_map[tree_id]
+            pp = tree_pp_map[tree_id]
+            tree_pcsps = tree.build_set_of_pcsps()
+            tree_pcsps = tree.build_set_of_pcsps()
+            for pcsp in tree_pcsps:
+                pcsp_pp_map[pcsp] += pp
+        return pcsp_pp_map
+
+    @staticmethod
+    def load_nni_list(nni_list_path):
+        git_commit = None
+        nni_list = []
+        nni_list_fp = open(nni_list_path, 'r')
+        for line in nni_list_fp.readlines():
+            fields = line.strip().split()
+            if not fields[0].startswith("#"):
+                nni_list.append(fields[0])
+            elif fields[0].startswith("#GIT_COMMIT"):
+                git_commit = fields[1]
+        nni_list_fp.close()
+        return git_commit, nni_list
+
+
+class Results:
+    def __init__(self, args):
+        self.data_ = {}
+        self.pre_data_ = {}
+        self.df_ = None
+        self.args = args
+        pass
+
+    def data_init(self):
+        self.data_ = {
```

Comment:
Would this not be easier with a Pandas dataframe?

Reply:
Yeah that makes sense.  Converted to dataframe.